programs/stake/src/stake_state.rs

```diff
@@ -613,15 +613,48 @@ pub fn delegate(
     }
 }
 
+fn deactivate_stake(
+    invoke_context: &InvokeContext,
+    stake: &mut Stake,
+    stake_flags: &mut StakeFlags,
+    epoch: Epoch,
+) -> Result<(), InstructionError> {
+    let stake_history = invoke_context.get_sysvar_cache().get_stake_history()?;
+
+    if invoke_context
+        .feature_set
+        .is_active(&feature_set::stake_redelegate_instruction::id())
+        && stake_flags.contains(StakeFlags::MUST_FULLY_ACTIVATE_BEFORE_DEACTIVATION_IS_PERMITTED)
```

Comment:
prefer not combining feature-gate activation test with feature-gated logic to simplify feature-gate removal

Reply:
Done.